### PR TITLE
Fixed Buy Button for Free NFT

### PR DESCRIPTION
**Pull request by faizhashmiRnS**
## Description
This PR should fix the issue Pylons-tech/pylons#1074 .

### CODE REVIEW

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

**Comment by StefanB90** [comment]
@faizhashmiRnS as discussed during standup, please add test cases. Thank you.

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart [hunk 1]
```diff
@@ -380,6 +346,74 @@ class _OwnerBottomDrawerState extends State<OwnerBottomDrawer> {
     );
   }
 
+  Widget getButtonContent(PurchaseItemViewModel viewModel) {
+    if (viewModel.nft.price == "0") {
```
[the remark this answers is not in the document]
Comment (kjawadDeveloper2) [reply wallet/lib/pages/purchase_item/purchase_item_screen.dart]: Solved the issue. @faizhashmiRnS  kindly add tests.

[the remark this answers is not in the document]
Comment (faizhashmiRnS) [reply wallet/lib/pages/purchase_item/purchase_item_screen.dart]: We are setting it to "0" when creating a Recipe in case of FreeDrop.

**Review by mustafapylons — APPROVED** [review]
Looks good to me.

File: wallet/lib/pages/purchase_item/purchase_item_screen.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (kjawadDeveloper2) [reply wallet/lib/pages/purchase_item/purchase_item_screen.dart]: we are but anyone can put amount in recipe when they use our sdk.

[the remark this answers is not in the document]
Comment (faizhashmiRnS) [reply wallet/lib/pages/purchase_item/purchase_item_screen.dart]: > we are but anyone can put amount in recipe when they use our sdk.

Got it, Thanks for the correction.

**Pull request merged**